modules/fedimint-ln-client/src/lib.rs

```diff
@@ -602,6 +653,86 @@ impl ClientModule for LightningClientModule {
             }
         }
     }
+
+    fn supports_backup(&self) -> bool {
+        true
+    }
+
+    async fn backup(
+        &self,
+        dbtx: &mut ModuleDatabaseTransaction<'_>,
+        executor: Executor<DynGlobalClientContext>,
+        api: DynGlobalApi,
+        module_instance_id: ModuleInstanceId,
+    ) -> anyhow::Result<Vec<u8>> {
+        let gateways = dbtx
+            .find_by_prefix(&LightningGatewayKeyPrefix)
+            .await
+            .map(|pair| pair.1)
+            .collect::<Vec<_>>()
+            .await;
+        let next_incoming_contract_index = dbtx
+            .get_value(&NextIncomingContractIndexKey)
+            .await
+            .unwrap_or(0);
+        let next_outgoing_contract_index = dbtx
+            .get_value(&NextOutgoingContractIndexKey)
+            .await
+            .unwrap_or(0);
+        let backup = LightningBackup {
+            gateways,
+            next_incoming_contract_index,
+            next_outgoing_contract_index,
+        };
+        Ok(backup.consensus_encode_to_vec()?)
+    }
+
+    async fn restore(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        module_instance_id: ModuleInstanceId,
+        executor: Executor<DynGlobalClientContext>,
+        api: DynGlobalApi,
+        snapshot: Option<&[u8]>,
+    ) -> anyhow::Result<()> {
+        // TODO: Need to look for active states here?
+
+        let snapshot = snapshot
+            .map(|mut s| LightningBackup::consensus_decode(&mut s, &Default::default()))
+            .transpose()?
+            .unwrap_or(LightningBackup::new_empty());
+        {
+            let module_dbtx = &mut dbtx.with_module_prefix(module_instance_id);
+            for gateway in snapshot.gateways {
+                module_dbtx
+                    .insert_new_entry(&LightningGatewayKey, &gateway)
+                    .await;
+            }
+
+            module_dbtx.insert_new_entry(
+                &NextIncomingContractIndexKey,
+                &snapshot.next_incoming_contract_index,
+            );
+            module_dbtx.insert_new_entry(
+                &NextOutgoingContractIndexKey,
+                &snapshot.next_outgoing_contract_index,
+            );
+        }
+
+        Ok(())
```

Comment:
This doesn't not attempt to "rewind the history" from the point of snapshot to the present. Is it planned&possible?

Reply:
In progress :)